snp_motion_planning/src/planning_server.cpp

```diff
@@ -27,7 +27,12 @@
 #include <tesseract_task_composer/planning/profiles/contact_check_profile.h>
 #include <tesseract_task_composer/planning/profiles/iterative_spline_parameterization_profile.h>
 #include <tesseract_task_composer/planning/profiles/min_length_profile.h>
+
+#if __has_include(<rclcpp/version.h>)  // ROS 2 Humble
```

Comment:
FYI, I think this macro is only available in c++17+, so you'll probably have to update the CMakeLists to reflect that. Also, should we be checking the version number inside this statement to make sure it is at least humble's version?

Reply:
That header file doesn't exist before humble, so it has to be at least humble by its presence.

Reply:
Also, apparently I didn't actually need it here, that was failing because of a CMake error. But I still call that macro elsewhere, how do I make ensure >=c++17?

Reply:
> That header file doesn't exist before humble, so it has to be at least humble by its presence.

I like this approach over manually creating some compile definition based on the string value of `ROS_VERSION`, but it feels a little hacky to just put it in there with that assumption. I was thinking we might do something like:

```c++
#if __has_include(<rclcpp/version.h>)
#include <rclcpp/version.h>
#endif
...
#if __has_include(<rclcpp/version.h>)
#if RCLCPP_VERSION_MAJOR > 16  // Find version where this change was introduced to make sure it exists
...
#endif
#endif
```

> Also, apparently I didn't actually need it here, that was failing because of a CMake error. But I still call that macro elsewhere, how do I make ensure >=c++17?

I'm not sure how that worked. AFAIK it only exists with c++17. You just need to set [`CMAKE_CXX_VERSION` to 17](https://github.com/ros-industrial-consortium/scan_n_plan_workshop/blob/9202d257f4798a56047ffe45582d76bd603061a2/snp_motion_planning/CMakeLists.txt#L5) in each package that uses it I think

Reply:
FYI I don't have any idea if `RCLCPP_VERSION_MAJOR > 16` is correct

Reply:
> FYI I don't have any idea if `RCLCPP_VERSION_MAJOR > 16` is correct. I just made that number up...

It's actually >=16, so you were very close. At least when I go to that header in my humble docker container I find the major version to be 16.